First iteration of WASM Clingo integration with webpanel view

CaptainUnbrauchbar opened the pull request:
😺

CaptainUnbrauchbar committed:
feat: replaced clingo binaries with wasm clingo

Co-authored-by: Richard Hegewald <richilino@users.noreply.github.com>

CaptainUnbrauchbar committed:
feat: :sparkles: first webview iteration

clingo output is now displayed in its own panel tab inside of a webview with answers divided into boxes

CaptainUnbrauchbar committed:
feat: :sparkles: added more features to webview

added welcome screen with tips, basic functioanlity of using path clingo

CaptainUnbrauchbar committed:
feat: :sparkles: finalized initial webpanel view

Co-authored-by: Richard Hegewald <richilino@users.noreply.github.com>

CaptainUnbrauchbar committed:
Merge pull request #23 from CaptainUnbrauchbar/webpanel-view

Replace console output with webpanel output

CaptainUnbrauchbar committed:
fix: :lock: improved semgrep findings

Co-authored-by: Richard Hegewald <richilino@users.noreply.github.com>

CaptainUnbrauchbar commented:
Attacking your own locally hosted webview panel be like:

![Bike-Fall](https://github.com/user-attachments/assets/bbabb235-7cb4-4c4c-beee-24227427aafa)

(Semgrep detections)

Also PATH Clingo has to be called in shell, theres no workaround yet.

CaptainUnbrauchbar committed:
feat: :sparkles: Added clingo running indicator and code to remove comments from input files

WASM Clingo doesnt support large input files so removing comments improves this problem

richilino committed:
refactor: :recycle: remove dead code, ran prettifier

richilino commented:
Added some broad refactorings, but we definitely need to improve the codebase further

CaptainUnbrauchbar committed:
Merge branch 'informaticup' into wasm-clingo-integration

CaptainUnbrauchbar committed:
feat: :sparkles: added progress indicator and option to cancel process to path clingo

path clingo runner now also displays an indicator that the process is running and shows a popup after 5 seconds so the user can cancel the process if it takes too long, this is subject to change
